services/src/main/java/org/keycloak/services/clientpolicy/executor/SecureRedirectUrisExecutor.java

```diff
@@ -0,0 +1,207 @@
+package org.keycloak.services.clientpolicy.executor;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.keycloak.OAuthErrorException;
+import org.keycloak.representations.idm.ClientPolicyExecutorConfigurationRepresentation;
+import org.keycloak.representations.idm.ClientRepresentation;
+import org.keycloak.services.clientpolicy.ClientPolicyContext;
+import org.keycloak.services.clientpolicy.ClientPolicyException;
+import org.keycloak.services.clientpolicy.context.AdminClientRegisterContext;
+import org.keycloak.services.clientpolicy.context.AdminClientUpdateContext;
+
+import java.net.InetAddress;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.UnknownHostException;
+import java.util.Collections;
+import java.util.List;
+
+public class SecureRedirectUrisExecutor implements ClientPolicyExecutorProvider<SecureRedirectUrisExecutor.Configuration> {
+    private Configuration configuration;
+
+    @Override
+    public void executeOnEvent(ClientPolicyContext context) throws ClientPolicyException {
+        ClientRepresentation proposedClient;
+        switch (context.getEvent()) {
+            case REGISTER:
+                proposedClient = ((AdminClientRegisterContext) context).getProposedClientRepresentation();
+                break;
+            case UPDATE:
+                proposedClient = ((AdminClientUpdateContext) context).getProposedClientRepresentation();
+                break;
+            default:
+                return;
+        }
+
+        for (String s : proposedClient.getRedirectUris()) {
+            validate(s);
+        }
+    }
+
+    void validate(String redirectUri) throws ClientPolicyException {
+        UriValidation validation;
+        try {
+            validation = new UriValidation(redirectUri);
+        } catch (URISyntaxException e) {
+            throw invalidRedirectUri("Invalid Redirect Uri: invalid uri syntax");
+        }
+
+        if (configuration.allowOpen) {
+            if (!"dev".equals(System.getProperty("kc.profile"))) {
```

Comment:
```suggestion
            if(!Environment.isDevProfile()){
```

Reply:
Hi @1-alex98 
Thanks, currently `Environment.isDevProfile` is from `keycloak-quarkus-server` module which is not included in the `keycloak-services` module.

I think it better keep as it is and wait for Keycloak team to check.